client/me/purchases/confirm-cancel-purchase/index.jsx

```diff
@@ -24,43 +24,36 @@ const ConfirmCancelPurchase = React.createClass( {
 		selectedSite: React.PropTypes.object
 	},
 
-	componentWillMount() {
-		recordPageView( 'confirm_cancel_purchase', this.props );
+	getInitialState() {
+		return {
+			html: null
+		};
 	},
 
-	componentWillReceiveProps( nextProps ) {
-		recordPageView( 'confirm_cancel_purchase', this.props, nextProps );
+	componentWillMount() {
+		this.loadEndpointForm();
+		recordPageView( 'confirm_cancel_purchase', this.props );
 	},
 
-	componentDidMount() {
-		if ( ! this.props.selectedPurchase.hasLoadedFromServer || ! this.props.selectedSite ) {
-			return;
+	componentDidUpdate( prevProps, prevState ) {
+		if ( ! prevState.html && this.state.html ) {
+			const container = React.findDOMNode( this.refs.root );
+
+			container.innerHTML = this.state.html;
+			initializeForm( {
+				form: container.querySelector( 'form' ),
+				onSubmit: this.handleSubmit,
+				selectedPurchase: getPurchase( this.props ),
+				selectedSite: this.props.selectedSite
+			} );
 		}
-
-		this.resetEndpointForm();
 	},
 
-	componentDidUpdate( prevProps ) {
-		if ( ! this.props.selectedPurchase.hasLoadedFromServer ) {
-			return;
-		}
-
-		if ( isEqual( prevProps.selectedPurchase.data, this.props.selectedPurchase.data ) &&
-				( prevProps.selectedSite.ID === this.props.selectedSite.ID ) ) {
-			return;
-		}
-
-		this.resetEndpointForm();
-	},
-
-	resetEndpointForm() {
-		const { selectedSite, selectedPurchase } = this.props;
-
-		loadEndpointForm( {
-			container: React.findDOMNode( this.refs.root ),
-			selectedSite,
-			selectedPurchase: selectedPurchase.data,
-			onSubmit: this.handleSubmit
+	loadEndpointForm() {
+		loadEndpointForm( getPurchase( this.props ), ( html ) => {
+			this.setState( {
+				html
+			} );
```

Comment:
I'm going to try this one @drewblaisdell. Thank you very much for this great tip. It looks much more clean and hides all those hacks in one place ;)